docs/src/rules/no-return-await.md

```diff
@@ -7,6 +7,13 @@ further_reading:
 - https://jakearchibald.com/2017/await-vs-return-vs-return-await/
 ---
 
+It is NOT recommended to use the `no-return-await` rule anymore because:
+
+* `return await` on a promise will not result in an extra microtask.
+* `return await` yields a better stack trace for debugging.
+
+The following documentation is kept for historical context, and is not necessarily accurate anymore.
+
```

Comment:
@kirkwaiblinger's done a ton of work on the tseslint side. I think it would be good to get Kirk's thoughts here if possible.